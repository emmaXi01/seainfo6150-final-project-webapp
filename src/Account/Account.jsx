import React from "react";
import PropTypes from "prop-types";
import styles from "./Account.module.css";

const Account = (props) => {
  return (
    <div className={styles.accountContainer}>
      <h4 className={styles.profile}>Public profile!</h4>
      <p className={styles.subtext}>Basic information about yourself</p>
      <label>
        Basics: 
        <div className={styles.username}>{props.username}</div>
      </label>        
    </div>
  );
};

Account.propTypes = {
  username: PropTypes.string.isRequired
};

export default Account;