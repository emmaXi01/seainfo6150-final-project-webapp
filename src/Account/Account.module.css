.accountContainer{
  width: 500px;
  margin: 5% auto 5%;
  padding: 50px 80px;
  border: 1px solid #ccc;
  border-radius: 10px;
}
  
.profile {
  display: block;
  text-align: center;
  color: #29303b;
  font-weight: 700;
  font-size: 20px;
}

.subtext {
  margin-top: -20px;
  text-align: center;
  margin-bottom: 50px;
}

.username {
  width: 100%;
  padding: 12px 10px;
  font-size: medium;
  margin: 8px 0; 
  border: 1px solid #ccc;
  box-sizing: border-box;
}

@media screen and (min-width: 400px) {
  .accountContainer{
    width: 300px;
  }
} 

@media screen and (min-width: 600px) {
  .accountContainer{
      width: 400px;
  }
}

@media screen and (min-width: 1150px) {
  .accountContainer{
      width: 600px;
  }
}