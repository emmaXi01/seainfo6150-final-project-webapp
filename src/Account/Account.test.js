import React from "react";
import { render } from "@testing-library/react";
import Account from "./Account.jsx";

describe("Account test", () => {
  it("renders the account component correctly", () => {
    const { container } = render(<Account username="Emma"/>);
    expect(container).toMatchSnapshot();
  });
});