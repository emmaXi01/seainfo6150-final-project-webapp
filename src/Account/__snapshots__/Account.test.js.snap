// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account test renders the account component correctly 1`] = `
<div>
  <div
    class="accountContainer"
  >
    <h4
      class="profile"
    >
      Public profile!
    </h4>
    <p
      class="subtext"
    >
      Basic information about yourself
    </p>
    <label>
      Basics:
      <div
        class="username"
      >
        Emma
      </div>
    </label>
  </div>
</div>
`;
